### DISCUSSION TITLE
Json Data Source using app settings value as url

### DISCUSSION THREAD

**Discussion by NikiSkaarup** [leekelleher/umbraco-contentment#332] [Idea]
Being able to set a json data source url via app settings / environment variables.

Would be really useful for when you use the same umbraco project for multiple client, thereby able to have different options per client.

**Comment by leekelleher** [comment]
Hi @NikiSkaarup, thank you for the suggestion.

Could you expand on how this could potentially work? 

My initial concern would be that if the configuration value (from app settings or environment variables) is applied to the JSON data-source - how would we deal with multiple instances (needing different configuration).

Developing your own custom data-source might mitigate potential issues of keeping the built-in JSON data-source simple/generic.

**Reply by NikiSkaarup** [reply]
Say we have these values in app settings / environment variables 
```json
{
  "DataSources": {
    "Colors": "http://localhost:5173/umbraco/colors",
    "Designs": "http://localhost:5173/umbraco/designs"
  },
}
```
```env
DataSources:Colors="http://localhost:5173/umbraco/colors"
DataSources:Design="http://localhost:5173/umbraco/designs"
```
and can then use `DataSources:Colors` to get the url instead of a static url for the whole umbraco project.

this would allow each deployment of a umbraco project to have a different url, as an example the umbraco backend has to point it at a headless frontend to get which options the frontend supports that the "editor" can pick

I think this would have to be a separate JSON data source, but most of the code can probably be shared between them

**Comment by leekelleher** [comment]
Closing this off, since it's been a few years, and I still believe this would be best suited as a custom data-source.
